app/controllers/api/v2/hosts_controller.rb

```diff
@@ -82,6 +82,12 @@ def show
       param_group :host, :as => :create
 
       def create
+        params[:host][:host_parameters_attributes] && params[:host] && params[:host][:host_parameters_attributes].each do |a|
```

Comment:
Testing `params[:host]` *after* accessing its `:host_parameters_attributes` member is the wrong order, as the hypothetical error would be on the first condition.